### ISSUE TITLE
🐛 Bug Report: `startTestBackend` fails with 1.28.0 update

### ISSUE THREAD

**Issue by drodil** [backstage/backstage#25311]
### 📜 Description

After updating to 1.28.0, the `startTestBackend` fails when providing root config mock as a feature.

### 👍 Expected behavior

The `startTestBackend` should allow adding `rootConfig` as part of the features array

### 👎 Actual Behavior with Screenshots

```
    console.error
      Failed to stop backend after tests, Error: Unexpected root lifecycle service implementation

      at error (../../../node_modules/@backstage/backend-test-utils/src/next/wiring/TestBackend.ts:353:19)
          at async Promise.all (index 0)
      at Object.<anonymous> (../../../node_modules/@backstage/backend-test-utils/src/next/wiring/TestBackend.ts:348:5)
```

```
 Unexpected plugin lifecycle service implementation
```

### 👟 Reproduction steps

Create a test that uses `startTestBackend`:

```ts
    const { server } = await startTestBackend({
      features: [
        myPlugin,
        mockServices.rootConfig.factory({
          data: {
            myPlugin: {
             config: 'value'
           },
          },
        }),
      ],
    });

    const response = await request(server).get('/api/myPlugin/health');
    expect(response.status).toBe(200);
    expect(response.body).toEqual({ status: 'ok' });
```

### 📃 Provide the context for the Bug.

https://github.com/backstage/backstage/blob/29a0f56831de683df148101a4acaa81971b2965f/packages/backend-test-utils/src/next/wiring/TestBackend.ts#L234

### 🖥️ Your Environment

OS:   Darwin 23.5.0 - darwin/arm64
node: v18.19.0
yarn: 1.22.22
cli:  0.26.7 (installed)
backstage:  1.28.0

Dependencies:
  @backstage/app-defaults                                            1.5.6
  @backstage/backend-app-api                                         0.7.5
  @backstage/backend-common                                          0.22.0
  @backstage/backend-defaults                                        0.2.18, 0.3.0
  @backstage/backend-dev-utils                                       0.1.4
  @backstage/backend-openapi-utils                                   0.1.12
  @backstage/backend-plugin-api                                      0.6.19
  @backstage/backend-tasks                                           0.5.24
  @backstage/backend-test-utils                                      0.4.0
  @backstage/catalog-client                                          1.6.5
  @backstage/catalog-model                                           1.5.0
  @backstage/cli-common                                              0.1.14
  @backstage/cli-node                                                0.2.6
  @backstage/cli                                                     0.26.7
  @backstage/config-loader                                           1.8.1
  @backstage/config                                                  1.2.0
  @backstage/core-app-api                                            1.12.6
  @backstage/core-compat-api                                         0.2.6
  @backstage/core-components                                         0.14.8
  @backstage/core-plugin-api                                         1.9.3
  @backstage/dev-utils                                               1.0.33
  @backstage/e2e-test-utils                                          0.1.1
  @backstage/errors                                                  1.2.4
  @backstage/eslint-plugin                                           0.1.8
  @backstage/frontend-plugin-api                                     0.6.6
  @backstage/integration-aws-node                                    0.1.12
  @backstage/integration-react                                       1.1.28
  @backstage/integration                                             1.12.0
  @backstage/plugin-api-docs                                         0.11.6
  @backstage/plugin-app-backend                                      0.3.68
  @backstage/plugin-app-node                                         0.1.19
  @backstage/plugin-auth-backend-module-atlassian-provider           0.2.0
  @backstage/plugin-auth-backend-module-aws-alb-provider             0.1.11
  @backstage/plugin-auth-backend-module-azure-easyauth-provider      0.1.2
  @backstage/plugin-auth-backend-module-bitbucket-provider           0.1.2
  @backstage/plugin-auth-backend-module-cloudflare-access-provider   0.1.2
  @backstage/plugin-auth-backend-module-gcp-iap-provider             0.2.14
  @backstage/plugin-auth-backend-module-github-provider              0.1.16
  @backstage/plugin-auth-backend-module-gitlab-provider              0.1.16
  @backstage/plugin-auth-backend-module-google-provider              0.1.16
  @backstage/plugin-auth-backend-module-guest-provider               0.1.5
  @backstage/plugin-auth-backend-module-microsoft-provider           0.1.14
  @backstage/plugin-auth-backend-module-oauth2-provider              0.2.0
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider        0.1.12
  @backstage/plugin-auth-backend-module-oidc-provider                0.2.0
  @backstage/plugin-auth-backend-module-okta-provider                0.0.12
  @backstage/plugin-auth-backend-module-onelogin-provider            0.1.0
  @backstage/plugin-auth-backend                                     0.22.6
  @backstage/plugin-auth-node                                        0.4.14
  @backstage/plugin-auth-react                                       0.1.3
  @backstage/plugin-catalog-backend-module-aws                       0.3.14
  @backstage/plugin-catalog-backend-module-github-org                0.1.14
  @backstage/plugin-catalog-backend-module-github                    0.6.2
  @backstage/plugin-catalog-backend-module-msgraph                   0.5.27
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model   0.1.17
  @backstage/plugin-catalog-backend-module-unprocessed               0.4.6
  @backstage/plugin-catalog-backend                                  1.23.0
  @backstage/plugin-catalog-common                                   1.0.24
  @backstage/plugin-catalog-graph                                    0.4.6
  @backstage/plugin-catalog-node                                     1.12.1
  @backstage/plugin-catalog-react                                    1.12.1
  @backstage/plugin-catalog-unprocessed-entities-common              0.0.2
  @backstage/plugin-catalog-unprocessed-entities                     0.2.5
  @backstage/plugin-catalog                                          1.21.0
  @backstage/plugin-devtools-backend                                 0.3.5
  @backstage/plugin-devtools-common                                  0.1.10
  @backstage/plugin-devtools                                         0.1.15
  @backstage/plugin-events-backend-module-github                     0.2.5
  @backstage/plugin-events-backend                                   0.3.6
  @backstage/plugin-events-node                                      0.3.5
  @backstage/plugin-home-react                                       0.1.14
  @backstage/plugin-home                                             0.7.5
  @backstage/plugin-kubernetes-common                                0.8.0
  @backstage/plugin-notifications-backend                            0.3.0
  @backstage/plugin-notifications-common                             0.0.3, 0.0.4
  @backstage/plugin-notifications-node                               0.1.4, 0.2.0
  @backstage/plugin-notifications                                    0.2.2
  @backstage/plugin-org                                              0.6.26
  @backstage/plugin-permission-backend                               0.5.43
  @backstage/plugin-permission-common                                0.7.14
  @backstage/plugin-permission-node                                  0.7.30
  @backstage/plugin-permission-react                                 0.4.23
  @backstage/plugin-proxy-backend                                    0.5.0
  @backstage/plugin-scaffolder-backend-module-azure                  0.1.11
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud        0.1.9
  @backstage/plugin-scaffolder-backend-module-bitbucket-server       0.1.9
  @backstage/plugin-scaffolder-backend-module-bitbucket              0.2.9
  @backstage/plugin-scaffolder-backend-module-confluence-to-markdown 0.2.20
  @backstage/plugin-scaffolder-backend-module-cookiecutter           0.2.43
  @backstage/plugin-scaffolder-backend-module-gerrit                 0.1.11
  @backstage/plugin-scaffolder-backend-module-gitea                  0.1.9
  @backstage/plugin-scaffolder-backend-module-github                 0.3.0
  @backstage/plugin-scaffolder-backend-module-gitlab                 0.4.1
  @backstage/plugin-scaffolder-backend-module-notifications          0.0.2
  @backstage/plugin-scaffolder-backend                               1.22.9
  @backstage/plugin-scaffolder-common                                1.5.3
  @backstage/plugin-scaffolder-node-test-utils                       0.1.5
  @backstage/plugin-scaffolder-node                                  0.4.5
  @backstage/plugin-scaffolder-react                                 1.9.0
  @backstage/plugin-scaffolder                                       1.21.0
  @backstage/plugin-search-backend-module-catalog                    0.1.25
  @backstage/plugin-search-backend-module-elasticsearch              1.5.0
  @backstage/plugin-search-backend-module-techdocs                   0.1.24
  @backstage/plugin-search-backend-node                              1.2.24
  @backstage/plugin-search-backend                                   1.5.10
  @backstage/plugin-search-common                                    1.2.12
  @backstage/plugin-search-react                                     1.7.12
  @backstage/plugin-search                                           1.4.12
  @backstage/plugin-signals-backend                                  0.1.5
  @backstage/plugin-signals-node                                     0.1.5
  @backstage/plugin-signals-react                                    0.0.3, 0.0.4
  @backstage/plugin-signals                                          0.0.7
  @backstage/plugin-techdocs-backend                                 1.10.6
  @backstage/plugin-techdocs-node                                    1.12.5
  @backstage/plugin-techdocs-react                                   1.2.5
  @backstage/plugin-techdocs                                         1.10.6
  @backstage/plugin-user-settings-backend                            0.2.18
  @backstage/plugin-user-settings-common                             0.0.1
  @backstage/plugin-user-settings                                    0.8.7
  @backstage/release-manifests                                       0.0.11
  @backstage/test-utils                                              1.5.6
  @backstage/theme                                                   0.5.6
  @backstage/types                                                   1.1.1
  @backstage/version-bridge                                          1.0.8

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by benjdlambert** [comment]
`@backstage/backend-app-api 0.7.5` should be `0.7.6`, is there something blocking updating some packages here?

**Comment by freben** [comment]
Yeah. [This is the check](https://github.com/backstage/backstage/blob/29a0f56831de683df148101a4acaa81971b2965f/packages/backend-app-api/src/wiring/BackendInitializer.ts#L362) in question. It used to perform a hard implementation check, but was updated in https://github.com/backstage/backstage/pull/24724 to be more "duck typed" to allow more flexibility.

**Comment by drodil** [comment]
@benjdlambert good catch, we had it in resolutions with `0.7.5` for some reason. Anyways updating it to `0.7.6` seems to fix this issue but not the #25310 so closing this one!
